runtime/src/lib.rs

```diff
@@ -485,44 +476,7 @@ mod validation_logic {
 					})
 				},
 			},
-			Categories::Shards(shards_script_info_struct) => {
-				let format = shards_script_info_struct.format;
-
-				// only support EDN for now
-				if format != ShardsFormat::Edn {
-					return false
-				}
-
-				// check trait exists and that we require proper references
-				let requiring = &shards_script_info_struct.requiring;
-				let require_check = requiring.iter().all(|trait_hash| {
-					// go thru all the things we reference, find shards scripts and check if they implement the trait
-					proto_references.iter().any(|proto_hash| {
-						if let Some(proto) = pallet_protos::Protos::<Runtime>::get(proto_hash) {
-							match proto.category {
-								Categories::Shards(shards_info) =>
-									shards_info.implementing.contains(trait_hash),
-								_ => false,
-							}
-						} else {
-							false
-						}
-					})
-				});
-
-				let implementing = &shards_script_info_struct.implementing;
-				let implement_check = implementing.iter().all(|&trait_hash| {
-					// `trait_protos` should always have a length of 1
-					let Some(trait_protos) = pallet_protos::ProtosByCategory::<Runtime>::get(Categories::Trait(Some(trait_hash))) else {
-						return false;
-					};
-					proto_references.iter().any(|proto_hash| {
-						trait_protos.contains(proto_hash)
-					})
-				});
-
-				require_check && implement_check
-			},
+			Categories::Shards(shards_script_info_struct) => shards_script_info_struct.format == ShardsFormat::Edn,
```

Comment:
@sinkingsugar  - should we validate that magic number of an edn file 😅 ?

Reply:
there is no such thing, but should just check it's valid utf8